### PR TITLE
Rex bugfixes

### PR DESCRIPTION
**Pull request by DethRaid**
Fix a bunch of bugs from me poorly using Rex

### CODE REVIEW

File: include/nova_renderer/util/utils.hpp [hunk 1]
```diff
@@ -34,6 +34,16 @@ namespace nova::renderer {
...
     bool ends_with(const std::string& string, const std::string& ending);
 
+    template<typename Arg>
+    uint32_t va_count(const Arg& arg) {
+        return 1;
+    }
+
+    template<typename Arg, typename... Args>
+    uint32_t va_count(const Arg arg, const Args&... args) {
+        return 1 + va_count(args);
```
Comment (cwfitzgerald) [inline include/nova_renderer/util/utils.hpp]: should be:

```cpp
template<typename... Args>
uint32_t va_count(const Args&... args) {
    return sizeof...(args);
}
```

File: src/filesystem/filesystem_helpers.cpp [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#include "nova_renderer/filesystem/filesystem_helpers.hpp"
+
+#include <stdarg.h>
+#include <string.h>
+
+namespace nova::filesystem {
+    void copy_path_parts_to_buffer(const rx_size offset, char* buffer, const rx::string& path_part) {
```
Comment (cwfitzgerald) [inline src/filesystem/filesystem_helpers.cpp]: Does this need to check if the buffer is large enough?

File: include/nova_renderer/util/utils.hpp (hunk 1 above)
Comment (cwfitzgerald) [reply include/nova_renderer/util/utils.hpp]: Tbh not sure this function needs to exist

File: src/rhi/vulkan/vulkan_render_device.cpp [hunk 3]
```diff
@@ -1248,8 +1248,8 @@ namespace nova::renderer::rhi {
                 return compute_family_index;
         }
 
-        logger(rx::log::level::k_error, "Unknown queue type &u", static_cast<uint32_t>(type));
-        return 999999; // Will probably cause a crash, which is actually what we want rn
+        logger(rx::log::level::k_error, "Unknown queue type %u", static_cast<uint32_t>(type));
+        return 999999; // Will cause a crash, which is actually what we want rn
```
Comment (cwfitzgerald) [inline src/rhi/vulkan/vulkan_render_device.cpp]: Then either assert false, or abort don't return garbage

**Review by cwfitzgerald — CHANGES_REQUESTED** [review]

File: include/nova_renderer/util/utils.hpp (hunk 1 above)
Comment (DethRaid) [reply include/nova_renderer/util/utils.hpp]: It actually doesn't. I thought I could be cool and make a variadic template to build a path out of an arbitrary number of strings, but then msvc hit the template recursion depth and I never tried ever again

File: src/rhi/vulkan/vulkan_render_device.cpp (hunk 3 above)
Comment (DethRaid) [reply src/rhi/vulkan/vulkan_render_device.cpp]: ugh fine I'll be reasonable

File: src/filesystem/filesystem_helpers.cpp (hunk 2 above)
Comment (DethRaid) [reply src/filesystem/filesystem_helpers.cpp]: Deleted the function

**Review by cwfitzgerald — APPROVED** [review]

**Pull request merged**